webknossos/webknossos/dataset/dataset.py

```diff
@@ -750,6 +755,118 @@ def add_layer_for_existing_files(
         layer.bounding_box = infer_bounding_box_existing_files(finest_mag_view)
         return layer
 
+    def add_layer_from_images(
+        self,
+        images: Union[str, "pims.FramesSequence", List[Union[str, PathLike]]],
+        ## add_layer arguments
+        layer_name: str,
+        category: LayerCategoryType = "color",
+        data_format: Union[str, DataFormat] = DEFAULT_DATA_FORMAT,
+        ## add_mag arguments
+        mag: Union[int, str, list, tuple, np.ndarray, Mag] = Mag(1),
+        chunk_size: Optional[Union[Vec3IntLike, int]] = None,
+        chunks_per_shard: Optional[Union[int, Vec3IntLike]] = None,
+        compress: bool = False,
```

Comment:
The advantage of this version is, that we can automatically infer the dtype and num_channels, which I find quite useful. Therefore, I'd keep it at the dataset level. I considered adding a `downsample` argument, but this can easily be done afterwards, e.g. `ds.add_layer_from_images(…).downsample()`.